Source/WebCore/rendering/BorderEdge.cpp

```diff
@@ -46,19 +46,24 @@ BorderEdge::BorderEdge(float edgeWidth, Color edgeColor, BorderStyle edgeStyle,
     m_flooredToDevicePixelWidth = floorf(edgeWidth * devicePixelRatio) / devicePixelRatio;
 }
 
-BorderEdges borderEdges(const RenderStyle& style, float deviceScaleFactor, bool includeLogicalLeftEdge, bool includeLogicalRightEdge)
+BorderEdges borderEdges(const RenderStyle& style, float deviceScaleFactor, bool setColorsToBlack, bool includeLogicalLeftEdge, bool includeLogicalRightEdge)
 {
     bool horizontal = style.isHorizontalWritingMode();
 
     auto constructBorderEdge = [&](float width, CSSPropertyID borderColorProperty, BorderStyle borderStyle, bool isTransparent, bool isPresent) {
-        return BorderEdge(width, style.visitedDependentColorWithColorFilter(borderColorProperty), borderStyle, isTransparent, isPresent, deviceScaleFactor);
+        auto color = setColorsToBlack ? Color::black : style.visitedDependentColorWithColorFilter(borderColorProperty);
+        return BorderEdge(width, color, borderStyle, isTransparent, isPresent, deviceScaleFactor);
+    };
+
+    auto borderIsTransparent = [&](bool borderIsTransparent) {
+        return !setColorsToBlack && borderIsTransparent;
     };
```

Comment:
Why not put this logic inside `constructBorderEdge` and avoid creating a second lambda? Just write `!setColorsToBlack && isTransparent` in there. The laziness of the `&&` operator isn’t happening here, so there is no performance benefit to do it this way.